tensorflow-framework/src/main/java/org/tensorflow/framework/losses/BinaryCrossentropy.java

```diff
@@ -0,0 +1,230 @@
+ *
+ * <p>Using <code>NONE</code> reduction type:
+ *
+ * <pre>
+ *    BinaryCrossentropy bce = new BinaryCrossentropy(tf, Reduction.NONE);
+ *    Operand&lt;TFloat32&gt; result = bce.call(labels, predictions);
+ *    // produces [0.916f, 0.714f]
+ * </pre>
+ */
+public class BinaryCrossentropy extends Loss {
+  public static final boolean FROM_LOGITS_DEFAULT = false;
+  public static final float LABEL_SMOOTHING_DEFAULT = 0.0f;
+
+  private final boolean fromLogits;
+  private final float labelSmoothing;
+
+  /**
+   * Creates a Binary Crossentropy Loss using {@link Class#getSimpleName()} as the loss name, {@link
+   * #FROM_LOGITS_DEFAULT} for fromLogits, {@link #LABEL_SMOOTHING_DEFAULT} for labelSmoothing and a
+   * Loss Reduction of {@link Loss#REDUCTION_DEFAULT}
+   *
+   * @param tf the TensorFlow Ops
+   */
+  public BinaryCrossentropy(Ops tf) {
+    this(tf, null, FROM_LOGITS_DEFAULT, LABEL_SMOOTHING_DEFAULT, REDUCTION_DEFAULT);
+  }
+
+  /**
+   * Creates a Binary Crossentropy loss using {@link Class#getSimpleName()} as the loss name, {@link
+   * #FROM_LOGITS_DEFAULT} for fromLogits, and {@link #LABEL_SMOOTHING_DEFAULT} for labelSmoothing
+   *
+   * @param tf the TensorFlow Ops
+   * @param reduction Type of Reduction to apply to the loss.
+   */
+  public BinaryCrossentropy(Ops tf, Reduction reduction) {
+    this(tf, null, FROM_LOGITS_DEFAULT, LABEL_SMOOTHING_DEFAULT, reduction);
+  }
+
+  /**
+   * Creates a Binary Crossentropy loss using using {@link Class#getSimpleName()} as the loss name,
+   * labelSmoothing of {@link #LABEL_SMOOTHING_DEFAULT}, a reduction of {@link
+   * Loss#REDUCTION_DEFAULT},
+   *
+   * @param tf the TensorFlow Ops
+   * @param fromLogits Whether to interpret predictions as a tensor of logit values
+   */
+  public BinaryCrossentropy(Ops tf, boolean fromLogits) {
+    this(tf, null, fromLogits, LABEL_SMOOTHING_DEFAULT, REDUCTION_DEFAULT);
+  }
+
+  /**
+   * Creates a Binary Crossentropy loss using labelSmoothing of {@link #LABEL_SMOOTHING_DEFAULT} a
+   * reduction of {@link Loss#REDUCTION_DEFAULT}.
+   *
+   * @param tf the TensorFlow Ops
+   * @param name the name of the loss
+   * @param fromLogits Whether to interpret predictions as a tensor of logit values
+   */
+  public BinaryCrossentropy(Ops tf, String name, boolean fromLogits) {
+    this(tf, name, fromLogits, LABEL_SMOOTHING_DEFAULT, REDUCTION_DEFAULT);
+  }
+
+  /**
+   * Creates a Binary Crossentropy loss using using {@link Class#getSimpleName()} as the loss name,
+   * and a reduction of {@link Loss#REDUCTION_DEFAULT}.
+   *
+   * @param tf the TensorFlow Ops
+   * @param fromLogits Whether to interpret predictions as a tensor of logit values
+   * @param labelSmoothing A number in the range, [0, 1]. When 0, no smoothing occurs. When &gt; 0,
+   *     compute the loss between the predicted labels and a smoothed version of the true labels,
+   *     where the smoothing squeezes the labels towards 0.5. Larger values of label_smoothing
```

Comment:
`label_smoothing` -> `labelSmoothing`, here and elsewhere in this file.

Reply:
OK